lib/main.py

```diff
@@ -27,57 +27,54 @@ async def lifespan(_app: FastAPI):
 
 
 APP = FastAPI(lifespan=lifespan)
-TASK_LIST: queue.Queue = queue.Queue(maxsize=100)
 
 
 class BackgroundProcessTask(threading.Thread):
     def run(self, *args, **kwargs):  # pylint: disable=unused-argument
+        nc = NextcloudApp()
+
+        provider_ids = set()
+        task_type_ids = set()
+        for chain_name, _ in chains.items():
+            provider_ids.add("llm2:" + chain_name)
+            (model, task) = chain_name.split(":", 2)
+            task_type_ids.add("core:text2text:" + task)
+
         while True:
-            task = TASK_LIST.get(block=True)
+            response = nc.providers.task_processing.next_task(list(provider_ids), list(task_type_ids))
+            if not isinstance(response, dict):
+                time.sleep(5)
+                continue
+
+            task = response["task"]
+            provider = response["provider"]
+
             try:
-                chain_name = task.get("chain")
+                chain_name = provider["name"][5:]
                 print(f"chain: {chain_name}", flush=True)
                 chain_load = chains.get(chain_name)
                 if chain_load is None:
-                    NextcloudApp().providers.text_processing.report_result(
-                        task["id"], error="Requested model is not available"
+                    NextcloudApp().providers.task_processing.report_result(
+                        task["id"], error_message="Requested model is not available"
                     )
                     continue
                 chain = chain_load()
                 print("Generating reply", flush=True)
                 time_start = perf_counter()
-                print(task.get("prompt"))
-                result = chain.invoke(task.get("prompt")).get("text")
+                print(task.get("input").get("input"))
+                result = chain.invoke(task.get("input").get("input")).get("text")
                 del chain
                 print(f"reply generated: {round(float(perf_counter() - time_start), 2)}s", flush=True)
                 print(result, flush=True)
-                NextcloudApp().providers.text_processing.report_result(
+                NextcloudApp().providers.task_processing.report_result(
                     task["id"],
-                    str(result),
+                    {"output": str(result)},
```

Comment:
and now this will be just `output=str(result)` ?

I am not sure about here, someone need to test this

Reply:
No, then it would have to be `output={"output": str(result)}`. The method parameter is called output, but the json also has a field called output.